### PR TITLE
[onert] Add unit tests for training Softmax op

### PR DESCRIPTION
**Pull request by ragmani**
This commit adds some tests to validate training of Softmax op.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
+
+    _context = std::make_unique<GenModelTrainContext>(cgen.finish());
+    _context->addTrainCase(
+      uniformTCD<float>({{{1, 3}}, {{2, 1}}},                                     // inputs
+                        {{{0, 1, 0, 0, 0, 0, 0, 0}}, {{0, 0, 0, 0, 0, 1, 0, 0}}}, // expected
+                        {0.1092f}                                                 // loss
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc:47]: The result of executing the same model with the same data on tensorflow:
```
Epoch 1/5
2/2 [==============================] - 0s 3ms/step - loss: 0.1094 - categorical_accuracy: 0.0000e+00
Epoch 2/5
2/2 [==============================] - 0s 1ms/step - loss: 0.1093 - categorical_accuracy: 0.5000
Epoch 3/5
2/2 [==============================] - 0s 1ms/step - loss: 0.1092 - categorical_accuracy: 0.5000
Epoch 4/5
2/2 [==============================] - 0s 1ms/step - loss: 0.1092 - categorical_accuracy: 0.5000
Epoch 5/5
2/2 [==============================] - 0s 1ms/step - loss: 0.1091 - categorical_accuracy: 0.5000
```

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    const float beta = 1.0f;
+    cgen.addOperatorFullyConnected({{input, weight, bias}, {fc_output}});
+    cgen.addOperatorSoftmax({{fc_output}, {output}}, beta);
+    cgen.setInputsAndOutputs({input}, {output});
+
+    float learning_rate = 0.01f;
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc:41]: If #13199 is merged first,
```suggestion
                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
    cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  int output = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_FLOAT32});
+  const float beta = 1.0f;
+  cgen.addOperatorSoftmax({{input}, {output}}, beta);
+  cgen.setInputsAndOutputs({input}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc:72]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Softmax.test.cc:94]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by zetwhite — APPROVED** [review]
LGTM :+1:
